<a href="https://colab.research.google.com/github/UadrianSP/am-turma5/blob/main/TratamentoAD_aula0901.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Pipeline de Analise de Dados [Census]

## Importação de Bibliotecas

*   Pandas
*   Numpy
*   MatPlotLib
*   SeaBorn



In [1]:
!pip install pandas
!pip install numpy
!pip install matplotlib
!pip install seaborn

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## Carregamento dos Dados da Base

In [4]:
base_census = pd.read_csv('/content/drive/MyDrive/am-t6/census.csv')

In [5]:
base_census

,age,workclass,final-weight,education,education-num,marital-status,occupation,relationship,race,sex,capital-gain,capital-loos,hour-per-week,native-country,income
0,39,State-gov,77516,Bachelors,13,Never-married,Adm-clerical,Not-in-family,White,Male,2174,0,40,United-States,<=50K
1,50,Self-emp-not-inc,83311,Bachelors,13,Married-civ-spouse,Exec-managerial,Husband,White,Male,0,0,13,United-States,<=50K
2,38,Private,215646,HS-grad,9,Divorced,Handlers-cleaners,Not-in-family,White,Male,0,0,40,United-States,<=50K
3,53,Private,234721,11th,7,Married-civ-spouse,Handlers-cleaners,Husband,Black,Male,0,0,40,United-States,<=50K
4,28,Private,338409,Bachelors,13,Married-civ-spouse,Prof-specialty,Wife,Black,Female,0,0,40,Cuba,<=50K
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
32556,27,Private,257302,Assoc-acdm,12,Married-civ-spouse,Tech-support,Wife,White,Female,0,0,38,United-States,<=50K
32557,40,Private,154374,HS-grad,9,Married-civ-spouse,Machine-op-inspct,Husband,White,Male,0,0,40,United-States,>50K
32558,58,Private,151910,HS-grad,9,Widowed,Adm-clerical,Unmarried,White,Female,0,0,40,United-States,<=50K
32559,22,Private,201490,HS-grad,9,Never-married,Adm-clerical,Own-child,White,Male,0,0,20,United-States,<=50K


In [6]:
base_census.describe()

,age,final-weight,education-num,capital-gain,capital-loos,hour-per-week
count,32561.000000,3.256100e+04,32561.000000,32561.000000,32561.000000,32561.000000
mean,38.581647,1.897784e+05,10.080679,1077.648844,87.303830,40.437456
std,13.640433,1.055500e+05,2.572720,7385.292085,402.960219,12.347429
min,17.000000,1.228500e+04,1.000000,0.000000,0.000000,1.000000
25%,28.000000,1.178270e+05,9.000000,0.000000,0.000000,40.000000
50%,37.000000,1.783560e+05,10.000000,0.000000,0.000000,40.000000
75%,48.000000,2.370510e+05,12.000000,0.000000,0.000000,45.000000
max,90.000000,1.484705e+06,16.000000,99999.000000,4356.000000,99.000000


## Caracteristicas da Base
- População amostral: 32.561
- Minorias importantes: 28 e 90, vol. Ed.=9
e 16,
ht= 40 e 99
- Media amostral: idade = 38.58, Ed = 10,
Gain. 1077.65, Loss = 87.30, ht= 40.44
- Desvio Padrão: [limites]: idade = 24.94, Ed. = 7.
51, Gain. = -6757.64 Loss= -315.66, ht= 28,9

# Pipeline de Divisão dos Dados [Eixos de Aprendizagem(Credit_Data)]

In [ ]:
X_credit = base_credit.iloc[:, 1:4].values

In [ ]:
X_credit

array([[6.61559251e+04, 5.90170151e+01, 8.10653213e+03],
       [3.44151540e+04, 4.81171531e+01, 6.56474502e+03],
       [5.73171701e+04, 6.31080495e+01, 8.02095330e+03],
       ...,
       [4.43114493e+04, 2.80171669e+01, 5.52278669e+03],
       [4.37560566e+04, 6.39717958e+01, 1.62272260e+03],
       [6.94365796e+04, 5.61526170e+01, 7.37883360e+03]])

In [ ]:
type(X_credit)

numpy.ndarray

In [ ]:
Y_credit = base_credit.iloc[:, 4].values

In [ ]:
Y_credit

array([0, 0, 0, ..., 1, 0, 0])

In [ ]:
type(Y_credit)

numpy.ndarray

In [ ]:
X_credit.shape, Y_credit.shape

((2000, 3), (2000,))

In [ ]:
X_credit[:,0].min(), X_credit[:,0].max()

(np.float64(20014.4894700497), np.float64(69995.6855783239))

In [ ]:
from sklearn.preprocessing import StandardScaler
scaler_credit = StandardScaler()
X_credit = scaler_credit.fit_transform(X_credit)

In [ ]:
X_credit[:,0].min(), X_credit[:,0].max()

(np.float64(-1.7676158019964077), np.float64(1.7220222385319197))

In [ ]:
X_credit

array([[ 1.45393393,  1.36539359,  1.20281942],
       [-0.76217555,  0.54267291,  0.69642695],
       [ 0.83682073,  1.67418453,  1.17471147],
       ...,
       [-0.07122592, -0.97447238,  0.35420081],
       [-0.11000289,  1.73938004, -0.92675625],
       [ 1.682986  ,  1.14918907,  0.96381038]])

## Amostras de Testes

In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
X_credit_train, X_credit_test, Y_credit_train, Y_credit_test = train_test_split(X_credit, Y_credit, test_size=0.25, random_state=0)

### Digestão de amostra do algoritmo de Naive Bayes

In [ ]:
X_credit_train.shape, Y_credit_train.shape

((1500, 3), (1500,))

### Gabarito de acurácia do algoritmo de Naive Bayes

In [ ]:
X_credit_test.shape, Y_credit_test.shape

((500, 3), (500,))

## Convertendo em Pickle

In [ ]:
import pickle

In [ ]:
with open('/content/drive/MyDrive/am-t6/credit.pkl', mode='wb') as f:
  pickle.dump([X_credit_train, Y_credit_train, X_credit_test, Y_credit_test], f)